frontend/public/components/cluster-settings/related-objects.tsx

```diff
@@ -0,0 +1,119 @@
+import { sortable } from '@patternfly/react-table';
+import { Table, TableRow, TableData, RowFunctionArgs } from '../factory';
+import { referenceForModel, K8sResourceKind, K8sKind } from '../../module/k8s';
+import { ResourceLink } from '../utils';
+
+const referenceFromGroupKind = (group: string = '', kind: string = '') => `${group}~${kind}`;
+
+const tableColumnClasses = [
+  classNames('col-lg-5', 'col-md-5', 'col-sm-4', 'col-xs-12'),
+  classNames('col-lg-2', 'col-md-2', 'col-sm-4', 'hidden-xs'),
+  classNames('col-lg-5', 'col-md-5', 'col-sm-4', 'hidden-xs'),
+];
+
+const Header = () => [
+  {
+    title: 'Name',
+    sortField: 'name',
+    transforms: [sortable],
+    props: { className: tableColumnClasses[0] },
+  },
+  {
+    title: 'Kind',
+    sortField: 'resource',
+    transforms: [sortable],
+    props: { className: tableColumnClasses[1] },
+  },
+  {
+    title: 'Namespace',
+    sortField: 'namespace',
+    transforms: [sortable],
+    props: { className: tableColumnClasses[2] },
+  },
+];
+
+const Row: React.FC<RowFunctionArgs> = ({ obj, index, key, style, customData }) => {
+  const { name, resource, namespace, group } = obj;
+  const uid = `${name}-${resource}-${namespace}-${group}`;
+  const model = customData[referenceFromGroupKind(group, resource)];
+
+  const gsv = model ? referenceForModel(model) : '';
+  return (
+    <TableRow id={uid} index={index} trKey={key} style={style}>
+      <TableData className={tableColumnClasses[0]}>
+        <ResourceLink kind={gsv} name={name} namespace={namespace} />
+      </TableData>
+      <TableData className={tableColumnClasses[1]}>{resource}</TableData>
+      <TableData className={tableColumnClasses[2]}>
+        {!_.isEmpty(namespace) ? <ResourceLink kind="Namespace" name={namespace} /> : ''}
+      </TableData>
+    </TableRow>
+  );
+};
+
+const RelatedObjects: React.FC<RelatedObjectsProps> = (props) => {
+  // All the models registered in the cluster
+  const { data } = props;
+  const models: K8sKind[] = _.values(
+    useSelector(({ k8s }) => k8s.getIn(['RESOURCES', 'models'])).toJS(),
+  );
+  const uniqueKinds = _.uniqBy(data, (item) => item.resource);
+  const relatedModels = models.reduce((acc, currModel) => {
+    // Check if current model is related to any of the data
+    // If so add a entry to acc that maps from group.resource to the resource
+    const relatedData = _.find(
+      uniqueKinds,
+      (item) =>
+        (currModel.kind === item.resource || currModel.plural === item.resource) &&
+        (currModel?.apiGroup ? currModel?.apiGroup === item?.group : true),
+    );
+    if (relatedData) {
+      acc[referenceFromGroupKind(relatedData?.group, relatedData.resource)] = currModel;
+    }
+    return acc;
+  }, {});
+
+  return (
+    <div className="co-m-pane__body">
+      <Table
+        {...props}
+        Header={Header}
+        Row={Row}
+        customData={relatedModels}
+        aria-label="Related Objects"
+        virtualize
```

Comment:
No need to virtualize this table, and we'd had a lot of display and performance problems with that. This list will never be very large.